File: app/src/main/java/com/babestudios/companyinfouk/ui/insolvencydetails/InsolvencyDetailsActivity.java
package com.babestudios.companyinfouk.ui.insolvencydetails;

import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;

import com.babestudios.companyinfouk.R;
import com.babestudios.companyinfouk.data.model.insolvency.InsolvencyCase;
import com.babestudios.companyinfouk.uiplugins.BaseActivityPlugin;
import com.babestudios.companyinfouk.utils.DividerItemDecorationWithSubHeading;
import com.google.gson.Gson;
import com.pascalwelsch.compositeandroid.activity.CompositeActivity;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;

public class InsolvencyDetailsActivity extends CompositeActivity{

	@BindView(R.id.toolbar)
	Toolbar toolbar;

	@BindView(R.id.insolvency_details_recycler_view)
	RecyclerView insolvencyDetailsRecyclerView;

	String insolvencyItemString;

	BaseActivityPlugin baseActivityPlugin = new BaseActivityPlugin();

	public InsolvencyDetailsActivity() {
		addPlugin(baseActivityPlugin);
	}

	@SuppressWarnings("ConstantConditions")
	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_insolvency_details);
		baseActivityPlugin.logScreenView(this.getLocalClassName());

		ButterKnife.bind(this);
		insolvencyItemString = getIntent().getStringExtra("insolvencyCase");
		Gson gson = new Gson();
		InsolvencyCase insolvencyCase = gson.fromJson(insolvencyItemString, InsolvencyCase.class);

		if (toolbar != null) {
			setSupportActionBar(toolbar);
			getSupportActionBar().setDisplayHomeAsUpEnabled(true);
			getSupportActionBar().setTitle(R.string.insolvency_details);
			toolbar.setNavigationOnClickListener(v -> onBackPressed());
		}
		createInsolvencyDetailsRecyclerView(insolvencyCase);
	}

	private void createInsolvencyDetailsRecyclerView(InsolvencyCase insolvencyCase) {
		LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false);
		insolvencyDetailsRecyclerView.setLayoutManager(linearLayoutManager);
		ArrayList<Integer> titlePositions = new ArrayList<>();
		titlePositions.add(0);
		titlePositions.add(insolvencyCase.dates.size() + 1);
		insolvencyDetailsRecyclerView.addItemDecoration(
				new DividerItemDecorationWithSubHeading(this, titlePositions));
		InsolvencyDetailsAdapter adapter = new InsolvencyDetailsAdapter(this, insolvencyCase.dates, insolvencyCase.practitioners);
		insolvencyDetailsRecyclerView.setAdapter(adapter);
	}

	@Override
	public void super_onBackPressed() {
		super.super_finish();
		super_overridePendingTransition(R.anim.left_slide_in, R.anim.left_slide_out);
	}
}

File: app/src/main/java/com/babestudios/companyinfouk/ui/insolvencydetails/InsolvencyDetailsActivity.kt
package com.babestudios.companyinfouk.ui.insolvencydetails

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.appcompat.widget.Toolbar
import butterknife.BindView
import butterknife.ButterKnife
import com.babestudios.companyinfouk.R
import com.babestudios.companyinfouk.data.model.insolvency.InsolvencyCase
import com.babestudios.companyinfouk.uiplugins.BaseActivityPlugin
import com.babestudios.base.view.DividerItemDecorationWithSubHeading
import com.google.gson.Gson
import com.pascalwelsch.compositeandroid.activity.CompositeActivity
import java.util.*

private const val INSOLVENCY_CASE = "com.babestudios.companyinfouk.ui.insolvency_case"

class InsolvencyDetailsActivity : CompositeActivity() {

	@JvmField
	@BindView(R.id.toolbar)
	internal var toolbar: Toolbar? = null

	@JvmField
	@BindView(R.id.insolvency_details_recycler_view)
	internal var insolvencyDetailsRecyclerView: RecyclerView? = null


	internal var baseActivityPlugin = BaseActivityPlugin()

	init {
		addPlugin(baseActivityPlugin)
	}

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_insolvency_details)
		baseActivityPlugin.logScreenView(this.localClassName)

		ButterKnife.bind(this)
		val insolvencyCase = intent.getParcelableExtra<InsolvencyCase>(INSOLVENCY_CASE)

		if (toolbar != null) {
			setSupportActionBar(toolbar)
			supportActionBar!!.setDisplayHomeAsUpEnabled(true)
			supportActionBar!!.setTitle(R.string.insolvency_details)
			toolbar!!.setNavigationOnClickListener { onBackPressed() }
		}
		createInsolvencyDetailsRecyclerView(insolvencyCase)
	}

	private fun createInsolvencyDetailsRecyclerView(insolvencyCase: InsolvencyCase) {
		val linearLayoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
		insolvencyDetailsRecyclerView!!.layoutManager = linearLayoutManager
		val titlePositions = ArrayList<Int>()
		titlePositions.add(0)
		titlePositions.add(insolvencyCase.dates.size + 1)
		insolvencyDetailsRecyclerView!!.addItemDecoration(
				DividerItemDecorationWithSubHeading(this, titlePositions))
		val adapter = InsolvencyDetailsAdapter(insolvencyCase.dates, insolvencyCase.practitioners)
		insolvencyDetailsRecyclerView!!.adapter = adapter
	}

	override fun super_onBackPressed() {
		super.super_finish()
		super_overridePendingTransition(R.anim.left_slide_in, R.anim.left_slide_out)
	}
}

fun Context.createInsolvencyDetailsIntent(insolvencyCase: InsolvencyCase): Intent {
	return Intent(this, InsolvencyDetailsActivity::class.java)
			.putExtra(INSOLVENCY_CASE, insolvencyCase)
}